lexos/models/tokenizer_model.py

```diff
@@ -0,0 +1,252 @@
+        file_col_dtm.insert(loc=0,
+                            column="Total",
+                            value=file_col_dtm.sum(axis=1))
+
+        file_col_dtm.insert(loc=1,
+                            column="Average",
+                            value=file_col_dtm["Total"] / len(labels))
+        return file_col_dtm
+
+    def _get_file_col_dtm_table(self) -> str:
+        """Get DTM with documents as columns and terms/characters as rows.
+
+        :return: An HTML formatted string that contains the desired DTM.
+        """
+        # Convert the HTML to beautiful soup object.
+        file_col_dtm_soup = BeautifulSoup(
+            self._get_file_col_dtm().round(3).to_html(
+                classes="table text-center table-bordered table-striped "
+                        "display no-wrap"),
+            "html.parser")
+
+        # Set the table style to 100% so it always takes up the space.
+        file_col_dtm_soup.find('table')['style'] = 'width: 100%'
+
+        # Return the beautiful soup object as a string.
+        return file_col_dtm_soup.prettify()
+
+    def _get_file_row_dtm(self) -> pd.DataFrame:
+        """Get DTM with documents as rows and terms/characters as columns.
+
+        :return: A pandas data frame that contains the DTM where each document
+                 is a row with total and average added to the original DTM.
+        """
+        # Check if empty DTM is received.
+        assert not self._doc_term_matrix.empty, EMPTY_DTM_MESSAGE
+
+        # Get temp file names.
+        labels = [self._id_temp_label_map[file_id]
+                  for file_id in self._doc_term_matrix.index.values]
+
+        # Get the main dtm, set proper column names and use labels as index.
+        file_row_dtm = self._doc_term_matrix
+        file_row_dtm.index = labels
+        file_row_dtm.columns.name = "Documents / Stats"
+
+        return file_row_dtm
+
+    def _get_file_row_dtm_table(self) -> str:
+        """Get DTM with documents as rows and terms/characters as columns.
+
+        :return: An HTML formatted string that contains the desired DTM.
+        """
+        # Get the file_row_dtm.
+        file_row_dtm = self._get_file_row_dtm()
+
+        # Convert the main dtm to beautiful soup object.
+        file_row_dtm_soup = BeautifulSoup(
+            file_row_dtm.to_html(
+                classes="table text-center table-bordered table-striped "
+                        "display no-wrap"),
+            "html.parser")
+
+        # Find the table head of the main dtm in order to insert stats info.
+        dtm_head = file_row_dtm_soup.find("thead")
+
+        # Form the total frame and set column name.
+        total_fame = pd.DataFrame(
+            columns=np.round(file_row_dtm.sum(axis=0), 3))
+        total_fame.columns.name = "Total"
+
+        # Convert the total frame to beautiful soup object.
+        total_soup = BeautifulSoup(total_fame.to_html(classes="table"),
+                                   "html.parser")
```

Comment:
I don't quite get why do you have to convert the frame to an html? html is much harder to manipulate than dataframe, since the data structure can be way to robust.

Reply:
I was trying to put two more rows in the table header : `total` and `average`. So that these two rows are fixed when page changes. But I guess there should be a better way.